server/etcdserver/api/v3compactor/periodic_test.go

```diff
@@ -172,3 +173,52 @@ func TestPeriodicPause(t *testing.T) {
 		t.Errorf("compact request = %v, want %v", a[0].Params[0], wreq.Revision)
 	}
 }
+
+func TestPeriodicSkipRevNotChange(t *testing.T) {
+	retentionMinutes := 5
+	retentionDuration := time.Duration(retentionMinutes) * time.Minute
+
+	fc := clockwork.NewFakeClock()
+	rg := &fakeRevGetter{testutil.NewRecorderStreamWithWaitTimout(10 * time.Millisecond), 0}
+	compactable := &fakeCompactable{testutil.NewRecorderStreamWithWaitTimout(10 * time.Millisecond)}
+	tb := newPeriodic(zap.NewExample(), fc, retentionDuration, rg, compactable)
+
+	tb.Run()
+	defer tb.Stop()
+
+	initialIntervals, intervalsPerPeriod := tb.getRetentions(), 10
+
+	// first compaction happens til 5 minutes elapsed
+	for i := 0; i < initialIntervals; i++ {
+		// every time set the same revision with 100
+		rg.SetRev(int64(100))
+		rg.Wait(1)
+		fc.Advance(tb.getRetryInterval())
+	}
+
+	// very first compaction
+	a, err := compactable.Wait(1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// first compaction the compact revision will be 100+1
+	expectedRevision := int64(100 + 1)
+	if !reflect.DeepEqual(a[0].Params[0], &pb.CompactionRequest{Revision: expectedRevision}) {
+		t.Errorf("compact request = %v, want %v", a[0].Params[0], &pb.CompactionRequest{Revision: expectedRevision})
+	}
+
+	// compaction doesn't happens at every interval since revision not change
+	for i := 0; i < 5; i++ {
+		for j := 0; j < intervalsPerPeriod; j++ {
+			rg.SetRev(int64(100))
+			rg.Wait(1)
+			fc.Advance(tb.getRetryInterval())
+		}
+
+		_, err := compactable.Wait(1)
```

Comment:
Should we check if any new compaction requests were recorded?

Reply:
yes, i added it.